core/ecschema-rpc/impl/src/SchemaRpcInterfaceImpl.ts

```diff
@@ -0,0 +1,134 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { SchemaRpcInterface } from "@bentley/schema-rpcinterface-common";
+import { IModelRpcProps, RpcManager } from "@bentley/imodeljs-common";
+import * as backend from "@bentley/imodeljs-backend";
+import { SchemaProps } from "@bentley/ecschema-metadata";
+import { ClientRequestContext } from "@bentley/bentleyjs-core";
+
+/**
+ * Defines the interface how the rows of the iModel query look like.
+ */
+interface SchemaNameRow {
+  schemaName: string;
+}
+
+/**
+ * Extends the iModelDb class with an overload of query that takes the row structure as
+ * generic parameter. This allows typed rows instead of any.
+ */
+type IModelDb = backend.IModelDb & {
+  query<T extends any>(ecsql: string): AsyncIterableIterator<T>;
+};
+
+/**
+ * Implementation of the SchemaRpcInterface.
+ */
+export class SchemaRpcInterfaceImpl extends SchemaRpcInterface {
```

Comment:
Impls should be marked `@internal`. The convention is to name the implementation of `FooRpcInterface` as `FooRpcImpl`.